File: multiple-languages/dotnet/AlibabaCloud.SDK.ROS.CDK.Directmail/AlibabaCloud/SDK/ROS/CDK/Directmail/IpfilterProps.cs
using Amazon.JSII.Runtime.Deputy;

#pragma warning disable CS0672,CS0809,CS1591

namespace AlibabaCloud.SDK.ROS.CDK.Directmail
{
    #pragma warning disable CS8618

    /// <summary>Properties for defining a `Ipfilter`.</summary>
    /// <remarks>
    /// See https://www.alibabacloud.com/help/ros/developer-reference/aliyun-directmail-ipfilter
    /// </remarks>
    [JsiiByValue(fqn: "@alicloud/ros-cdk-directmail.IpfilterProps")]
    public class IpfilterProps : AlibabaCloud.SDK.ROS.CDK.Directmail.IIpfilterProps
    {
        private object _ipAddress;

        /// <summary>Property ipAddress: The whitelist IP address.</summary>
        [JsiiProperty(name: "ipAddress", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
        public object IpAddress
        {
            get => _ipAddress;
            set
            {
                if (Amazon.JSII.Runtime.Configuration.RuntimeTypeChecking)
                {
                    switch (value)
                    {
                        case string cast_cd4240:
                            break;
                        case AlibabaCloud.SDK.ROS.CDK.Core.IResolvable cast_cd4240:
                            break;
                        case Amazon.JSII.Runtime.Deputy.AnonymousObject cast_cd4240:
                            // Not enough information to type-check...
                            break;
                        case null:
                            throw new System.ArgumentException($"Expected {nameof(value)} to be one of: string, {typeof(AlibabaCloud.SDK.ROS.CDK.Core.IResolvable).FullName}; received null", nameof(value));
                        default:
                            throw new System.ArgumentException($"Expected {nameof(value)} to be one of: string, {typeof(AlibabaCloud.SDK.ROS.CDK.Core.IResolvable).FullName}; received {value.GetType().FullName}", nameof(value));
                    }
                }
                _ipAddress = value;
            }
        }
    }
}

File: multiple-languages/go/alicloudroscdkdirectmail/datasource/IpfilterProps.go
package datasource


// Properties for defining a `Ipfilter`.
//
// See https://www.alibabacloud.com/help/ros/developer-reference/datasource-directmail-ipfilter
type IpfilterProps struct {
	// Property ipfilterId: The ID of the resource.
	IpfilterId interface{} `field:"required" json:"ipfilterId" yaml:"ipfilterId"`
	// Property refreshOptions: The refresh strategy for the datasource resource when the stack is updated.
	//
	// Valid values:
	// - Never: Never refresh the datasource resource when the stack is updated.
	// - Always: Always refresh the datasource resource when the stack is updated.
	// Default is Never.
	RefreshOptions interface{} `field:"optional" json:"refreshOptions" yaml:"refreshOptions"`
}

